crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -92,16 +95,28 @@ impl C8yMapperActor {
         messages: SimpleMessageBox<C8yMapperInput, C8yMapperOutput>,
         mqtt_publisher: LoggingSender<MqttMessage>,
         timer_sender: LoggingSender<SyncStart>,
+        ops_dir: PathBuf,
     ) -> Self {
         Self {
             converter,
             messages,
             mqtt_publisher,
             timer_sender,
+            ops_dir,
+            tedge_agent_status: false,
         }
     }
 
     async fn process_mqtt_message(&mut self, message: MqttMessage) -> Result<(), RuntimeError> {
+        // Send the init messages
+        if !self.tedge_agent_status && is_agent_health_status(&message) {
```

Comment:
Although there's nothing wrong in doing this here, I'd move this logic into the convertor as the `try_convert` function in it already has the list of all other message topics that it can handle. So, keeping the health-status check as well there would keep it consistent with the rest of the data. Of course, this message is not converted into anything. But you can alway return empty `Vec` from it.

Reply:
About checking the message `topic`, the `try_convert` will match generic topics and here we need to check for the specific topic name `tedge/health/tedge-agent`. Also, then we need to move the `ops_dir` to the converter to create the `operations files`.So, I feel it's better to do it here.